src/main/kotlin/org/eclipse/uprotocol/transport/builder/UAttributesBuilder.kt

```diff
@@ -36,16 +34,18 @@ class UAttributesBuilder
 /**
  * Construct the UAttributesBuilder with the configurations that are required for every payload transport.
  *
- * @param id       Unique identifier for the message.
- * @param type     Message type such as Publish a state change, RPC request or RPC response.
- * @param priority uProtocol Prioritization classifications.
- */ private constructor(private val id: UUID, private val type: UMessageType, private val priority: UPriority) {
-    private var ttl: Int? = null
-    private var token: String? = null
-    private var sink: UUri? = null
+ * @param uuid       Unique identifier for the message.
+ * @param messageType     Message type such as Publish a state change, RPC request or RPC response.
+ * @param uPriority uProtocol Prioritization classifications.
+ */ private constructor(
```

Comment:
Builders are not necessary in kotlin. Instead, we can add to the generated kotlin DSL to support the helper functions at the end of this class by leveraging context receivers: Here is an example for this class:
```kotlin
//UAttributesKtExt.kt
package org.eclipse.uprotocol.v1

import org.eclipse.uprotocol.uuid.factory.UuidFactory

context (UAttributesKt.Dsl)
@JvmSynthetic
fun forPublication(inPriority: UPriority) {
    id = UuidFactory.Factories.UPROTOCOL.factory().create()
    type = UMessageType.UMESSAGE_TYPE_PUBLISH
    priority = inPriority
}

context (UAttributesKt.Dsl)
@JvmSynthetic
fun forNotification(inPriority: UPriority, inSink: UUri) {
    id = UuidFactory.Factories.UPROTOCOL.factory().create()
    type = UMessageType.UMESSAGE_TYPE_RESPONSE
    priority = inPriority
    sink = inSink
}

context (UAttributesKt.Dsl)
@JvmSynthetic
fun forRequest(inPriority: UPriority, inSink: UUri, inTtl: Int) {
    id = UuidFactory.Factories.UPROTOCOL.factory().create()
    type = UMessageType.UMESSAGE_TYPE_REQUEST
    priority = inPriority
    sink = inSink
    ttl = inTtl
}

context (UAttributesKt.Dsl)
@JvmSynthetic
fun forResponse(inPriority: UPriority, inSink: UUri, reqId: UUID) {
    id = UuidFactory.Factories.UPROTOCOL.factory().create()
    type = UMessageType.UMESSAGE_TYPE_RESPONSE
    priority = inPriority
    sink = inSink
    reqid = reqId
}

```
In the tests, it would look like this: 
```kotlin
internal class UAttributesValidatorTest {
    @Test
    @DisplayName("test fetching validator for valid types")
    fun test_fetching_validator_for_valid_types() {
        val publish: UAttributesValidator = UAttributesValidator.getValidator(
            uAttributes {
                forPublication(UPriority.UPRIORITY_CS0)
            }
        )
        assertEquals("UAttributesValidator.Publish", publish.toString())
        val request: UAttributesValidator = UAttributesValidator.getValidator(
            uAttributes {
                forRequest(UPriority.UPRIORITY_CS4, uUri { }, 1000)
            }
        )
        assertEquals("UAttributesValidator.Request", request.toString())
        val response: UAttributesValidator = UAttributesValidator.getValidator(
            uAttributes{
                forResponse(
                    UPriority.UPRIORITY_CS4, uUri { }, UuidFactory.Factories.UPROTOCOL.factory().create()
                )
            }
        )
        assertEquals("UAttributesValidator.Response", response.toString())
    }

    @Test
    @DisplayName("Validate a UAttributes for payload that is meant to be published")
    fun test_validate_uAttributes_for_publish_message_payload() {
        val attributes = uAttributes { forPublication(UPriority.UPRIORITY_CS0) }
        val validator: UAttributesValidator = UAttributesValidator.Validators.PUBLISH.validator()
        val status: ValidationResult = validator.validate(attributes)
        assertTrue(status.isSuccess())
        assertEquals("", status.getMessage())
    }
    //...
}
```

Reply:
This builder class can then be removed. 
An issue can be created for this.

Reply:
Could you please go ahead and create an issue for this

Reply:
Sure, please find the issue here: https://github.com/eclipse-uprotocol/uprotocol-kotlin/issues/10